<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ElasticDlg</class>
 <widget class="QDialog" name="ElasticDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>567</width>
    <height>618</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Spurious Elastic and Inelastic Positions</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <property name="leftMargin">
    <number>4</number>
   </property>
   <property name="topMargin">
    <number>4</number>
   </property>
   <property name="rightMargin">
    <number>4</number>
   </property>
   <property name="bottomMargin">
    <number>4</number>
   </property>
   <property name="spacing">
    <number>2</number>
   </property>
   <item row="0" column="0" colspan="5">
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Inelastic Scan Positions</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>4</number>
       </property>
       <property name="topMargin">
        <number>4</number>
       </property>
       <property name="rightMargin">
        <number>4</number>
       </property>
       <property name="bottomMargin">
        <number>4</number>
       </property>
       <property name="spacing">
        <number>2</number>
       </property>
       <item row="1" column="2">
        <widget class="QPushButton" name="btnGenPositions">
         <property name="toolTip">
          <string>Generate scan positions.</string>
         </property>
         <property name="text">
          <string>Generate Positions...</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="btnDelPosition">
         <property name="toolTip">
          <string>Remove selected scan position.</string>
         </property>
         <property name="text">
          <string>Delete Position</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btnAddPosition">
         <property name="toolTip">
          <string>Add a new scan position.</string>
         </property>
         <property name="text">
          <string>Add Position</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="3">
        <widget class="QTableWidget" name="tablePositions">
         <property name="showDropIndicator" stdset="0">
          <bool>false</bool>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
         <column>
          <property name="text">
           <string>h (rlu)</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>k (rlu)</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>l (rlu)</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>ki (Å⁻¹)</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>kf (Å⁻¹)</string>
          </property>
         </column>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="btnLoadScan">
         <property name="toolTip">
          <string>Load the scan positions from a data file.</string>
         </property>
         <property name="text">
          <string>Load Scan...</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="btnSync">
         <property name="toolTip">
          <string>Get crystal and instrument configuration from Takin's main window.</string>
         </property>
         <property name="text">
          <string>Sync.</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Corresponding Elastic Positions</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="leftMargin">
        <number>4</number>
       </property>
       <property name="topMargin">
        <number>4</number>
       </property>
       <property name="rightMargin">
        <number>4</number>
       </property>
       <property name="bottomMargin">
        <number>4</number>
       </property>
       <property name="spacing">
        <number>2</number>
       </property>
       <item row="0" column="0">
        <widget class="QTextEdit" name="textResults">
         <property name="lineWrapMode">
          <enum>QTextEdit::NoWrap</enum>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Go to Pos.:</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QSpinBox" name="spinPosIdx">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Index of the scan position.</string>
     </property>
     <property name="prefix">
      <string/>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QPushButton" name="btnGotoElast1">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Go to elastic position for kf'=ki with given index.</string>
     </property>
     <property name="text">
      <string>kf' = ki</string>
     </property>
    </widget>
   </item>
   <item row="1" column="3">
    <widget class="QPushButton" name="btnGotoElast2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Go to elastic position for ki'=kf with given index.</string>
     </property>
     <property name="text">
      <string>ki' = kf</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QPushButton" name="btnGotoInel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Go to inelastic position with given index.</string>
     </property>
     <property name="text">
      <string>Original Position</string>
     </property>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="QPushButton" name="btnGotoInel1">
     <property name="toolTip">
      <string>Go to inelastic position for kf'=2ki with given index.</string>
     </property>
     <property name="text">
      <string>kf' = 2ki</string>
     </property>
    </widget>
   </item>
   <item row="2" column="3">
    <widget class="QPushButton" name="btnGotoInel2">
     <property name="toolTip">
      <string>Go to inelastic position for ki'=2kf with given index.</string>
     </property>
     <property name="text">
      <string>ki' = 2kf</string>
     </property>
    </widget>
   </item>
   <item row="2" column="4">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tablePositions</tabstop>
  <tabstop>btnAddPosition</tabstop>
  <tabstop>btnDelPosition</tabstop>
  <tabstop>btnGenPositions</tabstop>
  <tabstop>btnSync</tabstop>
  <tabstop>btnLoadScan</tabstop>
  <tabstop>textResults</tabstop>
  <tabstop>spinPosIdx</tabstop>
  <tabstop>btnGotoInel</tabstop>
  <tabstop>btnGotoElast1</tabstop>
  <tabstop>btnGotoElast2</tabstop>
  <tabstop>btnGotoInel1</tabstop>
  <tabstop>btnGotoInel2</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ElasticDlg</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ElasticDlg</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
